src/player.js

```diff
@@ -436,14 +436,17 @@ export default class Player extends FakeEventTarget {
    */
   play(): void {
     if (this._engine) {
-      this._playbackMiddleware.play(this._play.bind(this));
+      this._playbackMiddleware.play(() => this._play());
     } else if (this._loadingMedia) {
       // load media requested but the response is delayed
       Player._prepareVideoElement();
-      this._eventManager.listenOnce(this, CustomEventType.SOURCE_SELECTED, () => this.play());
+      if (this.config.plugins.ima) {
```

Comment:
Make this condition more generic and add it as a generic config.
The application can decide this, in this case if we have IMA or in future another ad plugin, we can set it in Kaltura-Player, or application that uses Kaltura player can even force this condition.

Reply:
let's make this callback name also more generic, as any middleware chain might want to use this. `_playAfterAsyncMiddleware` sounds short :satisfied: and concise